nservicebus/persistence/authoring-custom.md

```diff
@@ -113,50 +113,20 @@ The in-memory implementation of `ISagaPersister` can be found [here](https://git
-    /// Adds a new timeout.
-    /// </summary>
-    /// <param name="timeout">Timeout data.</param>
-    void Add(TimeoutData timeout);
-
-    /// <summary>
-    /// Removes the timeout if it hasn't been previously removed.
-    /// </summary>
-    /// <param name="timeoutId">The timeout id to remove.</param>
-    /// <param name="timeoutData">The timeout data of the removed timeout.</param>
-    /// <returns><c>true</c> it the timeout was successfully removed.</returns>
-    bool TryRemove(string timeoutId, out TimeoutData timeoutData);
-
-    /// <summary>
-    /// Removes the time by saga id.
-    /// </summary>
-    /// <param name="sagaId">The saga id of the timeouts to remove.</param>
-    void RemoveTimeoutBy(Guid sagaId);
-}
-```
+<!-- import PersistTimeoutsInterfaces -->
 
 The `TimeoutData` class holds timeout related data, like the `Time` it needs to fire at and the `SagaId` it is associated with. As a general rule, you should not use this class directly for persistence, but use another persistence class when possible and use the unique ID generation offered by the persistence you use.
 
 NServiceBus polls the persister for timeouts by calling `GetNextChunk`, and providing it with `DateTime startSlice` which specifies what is the last timeout it recieved in the previous call to this method, and then the persister should provide all timeouts that are due, meaning from that value to the current point in time. Some eventually consistent storages may require you to be innovative to make sure no timeouts are missed. Finally, the `nextTimeToRunQuery` needs to be set to tell NServiceBus when to next poll the persister for timeouts - usually this is set for the next known timeout after the current time. NServiceBus will automatically poll for timeouts again if it has reason to suspect new timeouts are available.
 
-The in-memory implementation of `IPersistTimeouts` can be seen [here](https://github.com/Particular/NServiceBus/blob/4.6.5/src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersistence.cs).
-
+In order to provide a custom timeout persister implementation: 
+- In versions 4.x and 5.x you need to implement interfaces `IPersistTimeouts` and `IPersistTimeoutsV2`. 
+    - The reference in-memory implementation of timeouts persistence for NServiceBus v4.x can be seen [here](https://github.com/Particular/NServiceBus/blob/support-4.4/src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersistence.cs). 
+    - The reference in-memory implementation of timeouts persistence for NServiceBus v5.x can be seen [here](https://github.com/Particular/NServiceBus/blob/support-5.0/src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersister.cs).
+ 
+The differences in interfaces definitions across versions are related to a patch fix preventing a potential message loss. You can find more details in the [issue description](https://github.com/Particular/NServiceBus/issues/2885).
```

Comment:
does that really add value to the doco? Shouldn't we just note that V2 is design to solve issues with non-transactional transports and replaces V1. V1 is just there to maintain backward compatibility.